bin/stalePullRequest.js

```diff
@@ -1,92 +1,131 @@
+        // If we're at the last page
+        if (!Cesium.defined(linkData) || !Cesium.defined(linkData.next)) {
+            return Promise.each(pullRequests, function (pullRequest) {
+                return stalePullRequest._processPullRequest(pullRequest, repositorySettings);
+            });
+        }
+        // Otherwise, request the next page
+        return requestPromise.get({
+            url: linkData.next.url,
+            headers: repositorySettings.headers,
+            json: true,
+            resolveWithFullResponse: true
+        }).then(processPage);
+    }
+
+    return requestPromise.get({
+        url: 'https://api.github.com/repos/' + repositoryName + '/pulls?state=open&base=master',
+        headers: repositorySettings.headers,
+        json: true,
+        resolveWithFullResponse: true
+    }).then(processPage);
+};
+
+stalePullRequest._processPullRequest = function (pullRequest, repositorySettings) {
+    var commentsUrl = pullRequest.comments_url;
+    return requestPromise.get({
+        url: commentsUrl,
+        headers: repositorySettings.headers,
+        json: true,
+        resolveWithFullResponse: true
+    })
+        .then(function (response) {
+            var linkData = parseLink(response.headers.link);
+            if (Cesium.defined(linkData)) {
+                commentsUrl = linkData.last.url;
+            }
+            return requestPromise.get({
+                url: commentsUrl,
+                headers: repositorySettings.headers,
+                json: true,
+            }).then(function (commentsJsonResponse) {
+                var lastComment = commentsJsonResponse[commentsJsonResponse.length - 1];
+                var foundStop = stalePullRequest.foundStopComment(commentsJsonResponse);
+                if (!foundStop && stalePullRequest.daysSince(new Date(lastComment.created_at)) >= repositorySettings.maxDaysSinceUpdate) {
+                    var template = repositorySettings.stalePullRequestTemplate;
+                    return requestPromise.post({
+                        url: commentsUrl,
+                        headers: repositorySettings.headers,
+                        body: {
+                            body: template({
+                                maxDaysSinceUpdate: repositorySettings.maxDaysSinceUpdate
+                            })
+                        },
+                        json: true
+                    });
+                }
+            });
+        });
+};
+
+stalePullRequest.daysSince = function (date) {
+    var msPerDay = 24 * 60 * 60 * 1000;
+    return (Date.now() - date.getTime()) / msPerDay;
+};
+
+stalePullRequest.foundStopComment = function (commentsJsonResponse) {
+    for(var i = 0; i < commentsJsonResponse.length; i++){
+        var comment = commentsJsonResponse[i].body.toLowerCase();
+        var userName = commentsJsonResponse[i].user.login;
+        if (userName !== 'cesium-concierge' && comment.indexOf('@cesium-concierge stop') !== -1) {

```

Comment:
I think we should have this command be configurable, or at least pulled out to where it can easily be found and edited, just in case.

Reply:
I'm planning on doing that as a separate PR so this doesn't get too big (see https://github.com/AnalyticalGraphicsInc/cesium-concierge/issues/110)